### PR TITLE
WebSockets Next: activate CDI request context only if needed

### PR DESCRIPTION
**Pull request by mkouba**
- related to #39148

### CODE REVIEW

**Comment by mkouba** [comment]
FYI @franz1981

File: docs/src/main/asciidoc/websockets-next-reference.adoc [hunk 1]
```diff
@@ -182,7 +182,10 @@ The session context remains active until the `@OnClose` method completes executi
 In cases where a WebSocket endpoint does not declare an `@OnOpen` method, the session context is still created.
 It remains active until the connection terminates, regardless of the presence of an `@OnClose` method.
 
-Methods annotated with `@OnTextMessage,` `@OnBinaryMessage,` `@OnOpen`, and `@OnClose` also have the request scope activated for the duration of the method execution (until it produced its result).
+Endpoint callbacks may also have the request context activated for the duration of the method execution (until it produced its result).
+By default, the request context is only activated if needed, i.e. if there is a request scoped bean , or a bean annotated with a security annotation (such as `@RolesAllowed`) in the dependency tree of the endpoint.
+However, it is possible to set the `quarkus.websockets-next.server.activate-request-context` config property to `always`.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/websockets-next-reference.adoc:187]: If I explicitly activate the request scope, would it work?

**Review by cescoffier — APPROVED** [review]
LGTM - wondering what happens if I use @ActivateRequestScope.

**Review by franz1981 — APPROVED** [review]
nice one @mkouba good idea! Activating it is indeed pretty costly!

**Review by michalvavrik — APPROVED** [review]

File: docs/src/main/asciidoc/websockets-next-reference.adoc (hunk 1 above)
Comment (mkouba) [reply docs/src/main/asciidoc/websockets-next-reference.adoc:187]: I've added a simple test...

**Pull request merged**